File: cpp/62048.cpp
#include <cstdio>
#include <iostream>

using namespace std;

int GCD(int a, int b){
    if ( a == 0 ) return b; 
    return GCD(b % a, a);
}

long long solution(int w,int h) {
    long long answer = 1;

    int gcd = GCD(w, h);
    answer = ((long)w * h) - ((long)w + h - gcd);
    return answer;
}

int main(){
    cout << solution(8, 12) << endl;
    return 0;
}

File: python3/62048.py
import math


def solution(w, h):
    return w*h - (w+h-math.gcd(w, h))


if __name__ == "__main__":
    t_case = []
    t_case.append({'w': 8, 'h': 12})
    t_case.append({'w': 8, 'h': 1})
    t_case.append({'w': 4, 'h': 1})
    t_case.append({'w': 8, 'h': 2})
    t_case.append({'w': 8, 'h': 8})

    for tc in t_case:
        print(solution(**tc))
